### ISSUE TITLE
Error loading PSD or PSB with 10 nested folders from SAI2 resaved with Photoshop

### ISSUE THREAD

**Issue by f2d** [Agamnentzar/ag-psd#77]
Hi.
I've seen that ag-psd now supports reading PSB and tried to parse some random files that I had around.

PSD and PSB test files created or resaved with SAI2 (ver.2021.07.06) or resaved with CSP (v1.8.7) load without error in ag-psd.
PSD and PSB test files resaved with Photoshop (tested with v20.0) both give the same error when loading in ag-psd.

My actual file had 700+ layers and 300+ folders nested to the limit allowed by SAI2 - 10 folder levels.
But the same error happens with a simpler setup - just 10 nested folders and one layer inside.
Maybe even simpler test is possible.
Nesting level itself is not the problem, error only shows on files resaved with Photoshop.

Just for the record, CSP shows a warning when saving such files:
![deeper than 5 levels - https;,,cdn discordapp com,attachments,851899006040408104,878284179333333022,unknown](https://user-images.githubusercontent.com/4068475/130282813-dc7a3ccf-68e3-486d-a922-2be9d134e61e.png)

Console error in Vivaldi (Chrome-based), latest version:
```
TypeError: Cannot read property 'children' of undefined
    at ag-psd-14.2.0-dist-bundle.js:5519
    at readSection (ag-psd-14.2.0-dist-bundle.js:5880)
    at readLayerInfo (ag-psd-14.2.0-dist-bundle.js:5480)
    at ag-psd-14.2.0-dist-bundle.js:5443
    at readSection (ag-psd-14.2.0-dist-bundle.js:5880)
    at Object.readPsd (ag-psd-14.2.0-dist-bundle.js:5442)
    at Object.readPsd (ag-psd-14.2.0-dist-bundle.js:5184)
```

Console error in Firefox, same error in version 56 and 90, only stack trace format is different, some clutter is omitted here:
```
TypeError: stack[(stack.length - 1)] is undefined
Stack trace:
readLayerInfo/<@ag-psd-14.2.0-dist-bundle.js:5519:17
readSection@ag-psd-14.2.0-dist-bundle.js:5880:18
readLayerInfo@ag-psd-14.2.0-dist-bundle.js:5480:5
readPsd/<@ag-psd-14.2.0-dist-bundle.js:5443:23
readSection@ag-psd-14.2.0-dist-bundle.js:5880:18
readPsd@ag-psd-14.2.0-dist-bundle.js:5442:5
readPsd@ag-psd-14.2.0-dist-bundle.js:5184:12
```

The bundle file is unmodified from https://cdn.jsdelivr.net/npm/ag-psd@14.2.0/dist/bundle.js

I'm attaching minimal 1x1px test files, error is triggered only by the Photoshop ones:
[test_1px_layer_10_folders_deep_PSD_and_PSB.zip](https://github.com/Agamnentzar/ag-psd/files/7023622/test_1px_layer_10_folders_deep_PSD_and_PSB.zip)

**Comment by Agamnentzar** [comment]
Fixed in `14.3.0` thank you for the test files

**Comment by f2d** [comment]
Works with my actual files so far, thanks for the quick fix.
